core/runtime/src/syscalls_impl_riscv.rs

```diff
@@ -0,0 +1,82 @@
+use libtock_platform::{OneArgMemop, RawSyscalls, YieldType, ZeroArgMemop};
+
+impl RawSyscalls for crate::TockSyscalls {
+    // This yield implementation is currently limited RISC-V versions without
+    // floating-point registers, as it does not mark them clobbered.
+    #[cfg(not(any(target_feature = "d", target_feature = "f")))]
+    fn raw_yield(r0_in: YieldType) -> u32 {
+        let mut r0 = r0_in as u32;
+        let mut _a4 = 0;
+        unsafe {
+            asm!("ecall",
+                 // x0 is a constant.
+                 lateout("x1") _, // Return address
```

Comment:
Nit: maybe use the ABI names like `ra`?

Reply:
In retrospect, I intended to include ABI names in all the comments. I intentionally used the x0...x31 naming because I really needed to go through every register individually. My latest commit adds those comments.

Is that fine with you? I realize it's a bit inconsistent with the rest of this file but it makes the comments on callee-saved registers easier to understand.